### PR TITLE
update openroad and allow for better control of heatmaps

### PR DESCRIPTION
**Pull request by gadfort**
Adds controls to heatmaps to set number of bins
Only emit heatmaps with data in them.
Always show clock tree.

### CODE REVIEW

File: siliconcompiler/tools/openroad/scripts/sc_procs.tcl [hunk 1]
```diff
@@ -116,10 +116,7 @@ proc sc_pin_placement { args } {
 ###########################
 
 proc sc_has_gui {} {
-  if {[expr [llength [info procs save_image]] > 0]} {
-      return true
-  }
-  return false
+  return [gui::supported]
```
Comment (troibe) [inline siliconcompiler/tools/openroad/scripts/sc_procs.tcl:119]: Is this a new attribute in openroad or comming from our siliconcompiler tcl?

File: siliconcompiler/tools/openroad/scripts/sc_write_images.tcl [hunk 2]
```diff
@@ -124,6 +166,7 @@ proc sc_image_routing_congestion {} {
     "Routing" \
     "routing_congestion.png" \
     "routing congestion"
+    0
```
Comment (troibe) [inline siliconcompiler/tools/openroad/scripts/sc_write_images.tcl:169]: So we are not using your new logic from the `sc_image_heatmap()` proc here?

**Review by troibe — APPROVED** [review]
Looks reasonable to me

File: siliconcompiler/tools/openroad/scripts/sc_write_images.tcl (hunk 2 above)
Comment (gadfort) [reply siliconcompiler/tools/openroad/scripts/sc_write_images.tcl:169]: the routing congestion is a little more limited so we cannot use that logic yet

**Pull request merged**